builder/dockerfile/copy_windows.go

```diff
@@ -67,21 +67,17 @@ func fixPermissionsWindows(source, destination, SID string) error {
 	sddlString := system.SddlAdministratorsLocalSystem
 	sddlString += "(A;OICI;GRGWGXRCWDSD;;;" + SID + ")"
 
-	securityDescriptor, err := winio.SddlToSecurityDescriptor(sddlString)
+	securityDescriptor, err := windows.SecurityDescriptorFromString(sddlString)
 	if err != nil {
 		return err
 	}
 
-	var daclPresent uint32
-	var daclDefaulted uint32
-	var dacl *byte
-
-	err = system.GetSecurityDescriptorDacl(&securityDescriptor[0], &daclPresent, &dacl, &daclDefaulted)
+	dacl, _, err := securityDescriptor.DACL()
 	if err != nil {
 		return err
 	}
 
-	return system.SetNamedSecurityInfo(windows.StringToUTF16Ptr(destination), windows.SE_FILE_OBJECT, windows.OWNER_SECURITY_INFORMATION|windows.DACL_SECURITY_INFORMATION, sid, nil, dacl, nil)
+	return windows.SetNamedSecurityInfo(destination, windows.SE_FILE_OBJECT, windows.OWNER_SECURITY_INFORMATION|windows.DACL_SECURITY_INFORMATION, sid, nil, dacl, nil)
```

Comment:
Wondering if `group` and `sacl` should also be set here (previously, they weren't;

```patch
diff --git a/builder/dockerfile/copy_windows.go b/builder/dockerfile/copy_windows.go
index 3f0ea32426..bff7b3aa87 100644
--- a/builder/dockerfile/copy_windows.go
+++ b/builder/dockerfile/copy_windows.go
@@ -76,8 +76,16 @@ func fixPermissionsWindows(source, destination, SID string) error {
        if err != nil {
                return err
        }
+       sacl, _, err := securityDescriptor.SACL()
+       if err != nil {
+               return err
+       }
+       group, _, err := securityDescriptor.Group()
+       if err != nil {
+               return err
+       }

-       return windows.SetNamedSecurityInfo(destination, windows.SE_FILE_OBJECT, windows.OWNER_SECURITY_INFORMATION|windows.DACL_SECURITY_INFORMATION, sid, nil, dacl, nil)
+       return windows.SetNamedSecurityInfo(destination, windows.SE_FILE_OBJECT, windows.OWNER_SECURITY_INFORMATION|windows.DACL_SECURITY_INFORMATION, sid, group, dacl, sacl)
 }

 func validateCopySourcePath(imageSource *imageMount, origPath, platform string) error {
```

Reply:
Should not be needed (it will inherit parent objects values by default)